cmake/Modules/FindTinyXML2.cmake

```diff
@@ -13,7 +13,26 @@ if(TinyXML2_FOUND)
       set(TINYXML2_LIBRARY tinyxml2)
     elseif(TARGET tinyxml2::tinyxml2)
       set(TINYXML2_LIBRARY tinyxml2::tinyxml2)
+    elseif(TinyXML2_FIND_REQUIRED)
+      message(FATAL_WARNING "Unable to determine target for TinyXML2")
     endif()
+    list(APPEND TinyXML2_TARGETS ${TINYXML2_LIBRARY})
+  else()
+    # Only perform that logic once
```

Comment:
When is this logic performed multiple time? Isn't the target appended multiple time to `TinyXML2_TARGETS` then?

Same below.